### ISSUE TITLE
--dry-run and --write-sql fail when a parameter of type \PDO::* or Connection::* is passed to addSql

### ISSUE THREAD

**Issue by sstalle** [doctrine/migrations#528]
Here is an example migration:
```php
<?php

namespace Application\Migrations;

use Doctrine\DBAL\Migrations\AbstractMigration;
use Doctrine\DBAL\Schema\Schema;
use Doctrine\DBAL\Connection;

class Version20170602195852 extends AbstractMigration
{
    /**
     * @param Schema $schema
     */
    public function up(Schema $schema)
    {
        $this->addSql(
            'SELECT id FROM users WHERE id IN (:ids)',
            ['ids' => [1, 2]],
            ['ids' => Connection::PARAM_INT_ARRAY]
        );
    }

    /**
     * @param Schema $schema
     */
    public function down(Schema $schema)
    {
    }
}
```

It can be executed successfully with `migrations:execute 20170602195852`. But adding a `--dry-run` or `--write-sql` option makes it throw an exception:

> [Doctrine\DBAL\DBALException]
>  Unknown column type "101" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgot to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information.

Here is a simple [test case](https://github.com/sstalle/migrations/pull/1/files).

**Comment by trickeyone** [comment]
Confirmed. I just ran into this issue myself. Migration scripts that were working before are now failed.

I've tracked the issue to `Doctrine\DBAL\Migrations\Version::formatParamsForOutput` line 467:
```php
private function formatParamsForOutput(array $params, array $types)
    {
        if (empty($params)) {
            return '';
        }

        $platform = $this->connection->getDatabasePlatform();
        $out = [];
        foreach ($params as $key => $value) {
            $type = isset($types[$key]) ? $types[$key] : 'string';
            $outval = Type::getType($type)->convertToDatabaseValue($value, $platform);
            $out[] = is_string($key) ? sprintf(':%s => %s', $key, $outval) : $outval;
        }

        return sprintf('with parameters (%s)', implode(', ', $out));
    }
```

The line ```$outval = Type::getType($type)->convertToDatabaseValue($value, $platform);` is the offending line. When calling `Doctrine\DBAL\Types\Type::getType`, it is expecting a string representation of the type, not the actual `PDO::PARAM_*` or `Connection::PARAM_*` value.

I did (hackishly) get it to execute by changing line 467 to:
```php
            $outval = $value;
            if (is_string($type)) {
                $outval = Type::getType($type)->convertToDatabaseValue($value, $platform);
            }
```

**Comment by mikeSimonson** [comment]
Fixed by #560
